internal/driver/deviceclient.go

```diff
@@ -50,7 +51,18 @@ func createCommandInfo(req *sdkModel.CommandRequest) (*CommandInfo, error) {
 	if err != nil {
 		return nil, err
 	}
-	length := calculateAddressLength(primaryTable, rawType)
+	var length uint16
+	switch req.Type {
+	case sdkModel.String:
+		l, err := strconv.Atoi(req.Attributes[STR_LENGTH])
```

Comment:
What is the STR_LENGTH? Is it an address length or string length?

For example, we have a string “Hello World!“ and the **string length** is 12. 
It takes 96 bits (12*8=96), so we should provide 6 register (96/16=6) for storing the  “Hello World!“ string. And the  **address length** is 6.

Reply:
ok, i`ll format that